--- aclocal.m4.old	2013-01-17 16:13:29.000000000 +0900
+++ aclocal.m4	2013-01-17 16:12:47.000000000 +0900
@@ -2013,20 +2013,7 @@
 # $2 = the with option name
 # $3 = the command to look for
 AC_DEFUN([FIND_GCC],[
-    if test "$TargetOS_CPP" = "darwin" &&
-       test "$XCodeVersion1" -eq 4 &&
-       test "$XCodeVersion2" -lt 2
-    then
-        # In Xcode 4.1, 'gcc-4.2' is the gcc legacy backend (rather
-        # than the LLVM backend). We prefer the legacy gcc, but in
-        # Xcode 4.2 'gcc-4.2' was removed.
-        FP_ARG_WITH_PATH_GNU_PROG([$1], [gcc-4.2], [gcc-4.2])
-    elif test "$windows" = YES
-    then
-        $1="$CC"
-    else
-        FP_ARG_WITH_PATH_GNU_PROG([$1], [$2], [$3])
-    fi
+    FP_ARG_WITH_PATH_GNU_PROG([$1], [$2], [$3])
     AC_SUBST($1)
 ])
 
