diff -ur ghc-7.6.1.orig/aclocal.m4 ghc-7.6.1/aclocal.m4
--- ghc-7.6.1.orig/aclocal.m4	2013-01-10 14:40:42.000000000 +0900
+++ ghc-7.6.1/aclocal.m4	2013-01-10 14:42:33.000000000 +0900
@@ -1792,7 +1792,7 @@
     dnl except we don't want to have to know what make is called. Sigh.
     rm -rf utils/ghc-pwd/dist-boot
     mkdir  utils/ghc-pwd/dist-boot
-    if ! "$WithGhc" -v0 -no-user-$GHC_PACKAGE_DB_FLAG -hidir utils/ghc-pwd/dist-boot -odir utils/ghc-pwd/dist-boot -stubdir utils/ghc-pwd/dist-boot --make utils/ghc-pwd/Main.hs -o utils/ghc-pwd/dist-boot/ghc-pwd
+    if ! "$WithGhc" -v0 -no-user-$GHC_PACKAGE_DB_FLAG -hidir utils/ghc-pwd/dist-boot -odir utils/ghc-pwd/dist-boot -stubdir utils/ghc-pwd/dist-boot --make utils/ghc-pwd/Main.hs -o utils/ghc-pwd/dist-boot/ghc-pwd -L/usr/lib
     then
         AC_MSG_ERROR([Building ghc-pwd failed])
     fi
diff -ur ghc-7.6.1.orig/rules/build-prog.mk ghc-7.6.1/rules/build-prog.mk
--- ghc-7.6.1.orig/rules/build-prog.mk	2013-01-10 14:40:45.000000000 +0900
+++ ghc-7.6.1/rules/build-prog.mk	2013-01-17 17:56:29.000000000 +0900
@@ -154,6 +154,16 @@
             $$(error Bad build stage)))),\
         $$$$(libraries/$$(dep)_dist-$(if $(filter 0,$3),boot,install)_v_LIB)))
 
+# dirty patch, sicne I don't know much about Makefile
+ifeq "$3" "0"
+ifeq "$$($1_$2_LINK_WITH_GCC)" "NO"
+$1/$2/build/tmp/$$($1_$2_PROG) : $$($1_$2_v_HS_OBJS) $$($1_$2_v_C_OBJS) $$($1_$2_v_S_OBJS) $$($1_$2_OTHER_OBJS) | $$$$(dir $$$$@)/.
+	$$(call cmd,$1_$2_HC) -o $$@ $$($1_$2_v_ALL_HC_OPTS) $$(LD_OPTS) -L/usr/lib $$($1_$2_GHC_LD_OPTS) $$($1_$2_v_HS_OBJS) $$($1_$2_v_C_OBJS) $$($1_$2_v_S_OBJS) $$($1_$2_OTHER_OBJS) $$(addprefix -l,$$($1_$2_EXTRA_LIBRARIES))
+else
+$1/$2/build/tmp/$$($1_$2_PROG) : $$($1_$2_v_HS_OBJS) $$($1_$2_v_C_OBJS) $$($1_$2_v_S_OBJS) $$($1_$2_OTHER_OBJS) | $$$$(dir $$$$@)/.
+	$$(call cmd,$1_$2_CC)" -o $$@ $$($1_$2_v_ALL_CC_OPTS) $$(LD_OPTS) -L/usr/lib $$($1_$2_v_HS_OBJS) $$($1_$2_v_C_OBJS) $$($1_$2_v_S_OBJS) $$($1_$2_OTHER_OBJS) $$($1_$2_v_EXTRA_CC_OPTS) $$(addprefix -l,$$($1_$2_EXTRA_LIBRARIES))
+endif
+else
 ifeq "$$($1_$2_LINK_WITH_GCC)" "NO"
 $1/$2/build/tmp/$$($1_$2_PROG) : $$($1_$2_v_HS_OBJS) $$($1_$2_v_C_OBJS) $$($1_$2_v_S_OBJS) $$($1_$2_OTHER_OBJS) | $$$$(dir $$$$@)/.
 	$$(call cmd,$1_$2_HC) -o $$@ $$($1_$2_v_ALL_HC_OPTS) $$(LD_OPTS) $$($1_$2_GHC_LD_OPTS) $$($1_$2_v_HS_OBJS) $$($1_$2_v_C_OBJS) $$($1_$2_v_S_OBJS) $$($1_$2_OTHER_OBJS) $$(addprefix -l,$$($1_$2_EXTRA_LIBRARIES))
@@ -161,6 +171,7 @@
 $1/$2/build/tmp/$$($1_$2_PROG) : $$($1_$2_v_HS_OBJS) $$($1_$2_v_C_OBJS) $$($1_$2_v_S_OBJS) $$($1_$2_OTHER_OBJS) | $$$$(dir $$$$@)/.
 	$$(call cmd,$1_$2_CC)" -o $$@ $$($1_$2_v_ALL_CC_OPTS) $$(LD_OPTS) $$($1_$2_v_HS_OBJS) $$($1_$2_v_C_OBJS) $$($1_$2_v_S_OBJS) $$($1_$2_OTHER_OBJS) $$($1_$2_v_EXTRA_CC_OPTS) $$(addprefix -l,$$($1_$2_EXTRA_LIBRARIES))
 endif
+endif
 
 # Note [lib-depends] if this program is built with stage1 or greater, we
 # need to depend on the libraries too.  NB. since $(ALL_STAGE1_LIBS) and
diff -ur ghc-7.6.1.orig/utils/ghc-cabal/ghc.mk ghc-7.6.1/utils/ghc-cabal/ghc.mk
--- ghc-7.6.1.orig/utils/ghc-cabal/ghc.mk	2013-01-10 14:40:45.000000000 +0900
+++ ghc-7.6.1/utils/ghc-cabal/ghc.mk	2013-01-10 14:43:27.000000000 +0900
@@ -35,7 +35,8 @@
 	       -ilibraries/Cabal/Cabal \
 	       -ilibraries/filepath \
 	       -ilibraries/hpc \
-	       $(utils/ghc-cabal_dist_EXTRA_HC_OPTS)
+	       $(utils/ghc-cabal_dist_EXTRA_HC_OPTS) \
+	       -L/usr/lib
 	"$(TOUCH_CMD)" $@
 
 # touch is required, because otherwise if mkdirhier is newer, we
diff -ur ghc-7.6.1.orig/utils/ghc-pkg/ghc.mk ghc-7.6.1/utils/ghc-pkg/ghc.mk
--- ghc-7.6.1.orig/utils/ghc-pkg/ghc.mk	2013-01-10 14:40:45.000000000 +0900
+++ ghc-7.6.1/utils/ghc-pkg/ghc.mk	2013-01-10 14:44:31.000000000 +0900
@@ -66,7 +66,8 @@
 	       -ilibraries/filepath \
 	       -ilibraries/hpc \
 	       -ilibraries/binary/src \
-	       -ilibraries/bin-package-db
+	       -ilibraries/bin-package-db \
+	       -L/usr/lib
 
 
 utils/ghc-pkg/Version.hs: mk/project.mk
