diff -ur nvi-1.81.6.orig/ex/ex_script.c nvi-1.81.6/ex/ex_script.c
--- nvi-1.81.6.orig/ex/ex_script.c	2013-01-22 20:54:53.000000000 +0900
+++ nvi-1.81.6/ex/ex_script.c	2013-01-22 21:16:14.000000000 +0900
@@ -23,7 +23,7 @@
 #include <sys/select.h>
 #endif
 #include <sys/stat.h>
-#ifdef HAVE_SYS5_PTY
+#if 0
 #include <sys/stropts.h>
 #endif
 #include <sys/time.h>
@@ -664,7 +664,7 @@
 	F_CLR(gp, G_SCRWIN);
 }
 
-#ifdef HAVE_SYS5_PTY
+#if 0
 static int ptys_open __P((int, char *));
 static int ptym_open __P((char *));
 
